### PR TITLE
Disable experimental flag on code for 5.0.0

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Remove experimental flag on breaking changes for 5.0.0

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 41 | 37 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 45 | 54 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 62 | 67 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 34 | 49 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2248 | 2308 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2194 | 2301 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 34 | 49 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 37 | 37 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2596 | 2978 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 52 | 123 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 76 | 97 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 208 | 142 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 48 | 56 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 140 | 184 | 3003 |
|  connections.Connection | 14049 | 14049 | 51 | 58 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 102 | 112 | 4516 |
|  create.SimpleMutation | 7 | 7 | 49 | 58 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 21 | 22 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 186 | 200 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 141 | 159 | 1183 |
|  2871.NestedRelationshipFilter | 23093 | 23071 | 50 | 59 | 4438 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 35 | 54 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 78 | 105 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 32 | 44 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 51 | 62 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 21 | 20 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 41 | 39 | 2174 |
|  query.QueryWhere | 9676 | 9690 | 30 | 45 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9843 | 9857 | 45 | 50 | 2161 |
|  query.Nested | 10095502 | 10095502 | 6479 | 7206 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6823 | 6611 | 2004000 |
|  query.OrFilterOnRelationships | 43643 | 43831 | 134 | 271 | 2031 |
|  query.OrFilterOnRelationshipsAndNested | 36723 | 36693 | 171 | 302 | 2031 |
|  query.QueryWithNestedIn | 13632 | 13609 | 45 | 68 | 1305 |
|  query.NestedConnectionWhere | 9801 | 9801 | 51 | 73 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10006 | 9848 | 72 | 95 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18962 | 18752 | 129 | 159 | 1652 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 161 | 182 | 1134 |
|  query.Fulltext | 64 | 64 | 26 | 46 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 38 | 56 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 55 | 48 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 80 | 70 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14188 | 14185 | 63 | 69 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14188 | 14185 | 89 | 106 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14582 | 14792 | 106 | 111 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14432 | 14496 | 133 | 105 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 74 | 85 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14501 | 14505 | 100 | 103 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 39 | 55 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 78 | 87 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 64 | 77 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 54 | 66 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 54 | 57 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 290 | 312 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 236 | 294 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 89 | 119 | 2002 |

</details>
        
Old Schema Generation: 24.800s
Schema Generation: 26.310s
Old Subgraph Schema Generation: 27.533s
Subgraph Schema Generation: 29.673s

File: packages/graphql/src/translate/queryAST/factory/OperationFactory.ts [hunk 1]
```diff
@@ -287,7 +287,7 @@ export class OperationsFactory {
         } else {
             // if typename is allowed and therefore _on is disabled we can compute only the shared filter without recomputing the filters for each concrete entity
             // if typename filters are allowed we are getting rid of the _on and the implicit typename filter.
-            const typenameFilterAllowed = this.experimental && isInterfaceEntity(entity);
+            const typenameFilterAllowed = isInterfaceEntity(entity);
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/factory/OperationFactory.ts:290]: I will change the name from `typenameFilterAllowed` to `isInterface` as by removing the experimental flag any interface supports typename filters.

File: packages/graphql/tests/schema/directives/relationship-nested-operations.test.ts [hunk 2]
```diff
@@ -5233,6 +5235,7 @@ describe("Relationship nested operations", () => {
                   movies(options: MovieOptions, where: MovieWhere): [Movie!]!
                   moviesAggregate(where: MovieWhere): MovieAggregateSelection!
                   moviesConnection(after: String, first: Int, sort: [MovieSort], where: MovieWhere): MoviesConnection!
+                  people(options: QueryOptions, where: PersonWhere): [Person!]!
```
Comment (MacondoExpress) [inline packages/graphql/tests/schema/directives/relationship-nested-operations.test.ts:5238]: Do we have introduced top-level union queries? can you double-check this?

**Review by MacondoExpress — COMMENTED** [review]
Left some comments, overall looks good, can you just double-check the top-level union queries on the schema tests?

File: packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts [hunk 3]
```diff
@@ -87,17 +87,6 @@ export function createRelationshipFields({
...
-                relationship: relationshipAdapter,
-                composeNode,
-                schemaComposer,
-                userDefinedFieldDirectives,
-            });
-
-            return;
-        }
-
         // TODO: new way
```
[the remark this answers is not in the document]
Comment (a-alle) [reply packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts:88]: the comment is related to the schema generation refactor and is still relevant, albeit abstract

File: packages/graphql/tests/schema/directives/relationship-nested-operations.test.ts (hunk 2 above)
Comment (a-alle) [reply packages/graphql/tests/schema/directives/relationship-nested-operations.test.ts:5238]: yes, we have simple queries for unions implemented. There is no sorting supported on them.

File: packages/graphql/src/schema/make-augmented-schema.ts [hunk 4]
```diff
@@ -455,36 +453,34 @@ function makeAugmentedSchema({
                     includeRelationships: true,
                 },
             });
-            if (experimental) {
-                // TODO: mirror everything on interfaces target of relationships
-                // TODO [top-level-abstract-types-filtering]: _on should contain also implementing interface types?
-                withWhereInputType({
+            // TODO: mirror everything on interfaces target of relationships
+            // TODO [top-level-abstract-types-filtering]: _on should contain also implementing interface types?
```
Comment (a-alle) [inline packages/graphql/src/schema/make-augmented-schema.ts]: this comment is no longer relevant as there is no `_on` in filtering anymore

File: packages/graphql/src/schema/validation/custom-rules/warnings/limit-max-can-be-bypassed.ts [hunk 5]
```diff
@@ -19,25 +19,20 @@
...
+import { getPathToNode } from "../utils/path-parser";
+import { parseArgumentToInt } from "../utils/utils";
 
-export function WarnIfAMaxLimitCanBeBypassedThroughInterface(experimental: boolean) {
-    if (!experimental) {
-        return (): ASTVisitor => {
-            return {};
-        };
-    }
+export function WarnIfAMaxLimitCanBeBypassedThroughInterface(_experimental: boolean) {
```
Comment (a-alle) [inline packages/graphql/src/schema/validation/custom-rules/warnings/limit-max-can-be-bypassed.ts]: experimental argument can be removed

File: packages/graphql/tests/integration/experimental/aggegations/aggregation-interfaces-field-level.int.test.ts [hunk 6]
```diff
@@ -69,7 +69,7 @@ describe("Interface Field Level Aggregations", () => {
             }
         `;
 
-        neoSchema = new Neo4jGraphQL({ typeDefs, experimental: true });
+        neoSchema = new Neo4jGraphQL({ typeDefs, experimental: false });
```
Comment (a-alle) [inline packages/graphql/tests/integration/experimental/aggegations/aggregation-interfaces-field-level.int.test.ts]: experimental argument can be left out

**Pull request merged**